### ISSUE TITLE
ENH: Generic device configuration wrapper 

### ISSUE THREAD

**Issue by ldorofeeva** [bluesky/bluesky#1762]
# Create a generic device configuration wrapper

## Expected Behavior
The wrapper should act similar to `configure_count_time_wrapper` but instead of configuring a fixed-name component `count_time` it should take a dictionary `{<component_name>: <component_value>}` as a second parameter.
Logic (for each device) is as follows: for each `<name>, <value>` pair from the dictionary, if the device has an `ophyd.Signal`-type attribute with given name, the attribute should be set to given `<value>`.
Similarly to `configure_count_time_wrapper`, original values of all components are cached and restored in a `finalize_wrapper`

## Current Behavior
Currently there is no generic wrapper to configure device components with arbitrary names

## Possible Solution
Implement a generic wrapper as per logic described above
Basically, it is already implemented as a part of CLS (work-in-progress) bluesky project, but I thought to add it to the codebase as other bluesky users may find it convenient.

## Context
I would like to use this wrapper to configure various components of the devices used in plans in bulk (i.e. detectors acquisition mode, or dwell time that may not be named `count_time`, motors velocity and acceleration etc) and have them automatically reset to initial state at the end.
I could also foresee a possible use of it as a RE preprocessor to configure devices according to session/persistent metadata (i.e. file paths for file-writing detectors), but this is yet an area to explore.

_Please assign myself._
